x/supplier/keeper/prune_supplier_service_config_history.go

```diff
@@ -0,0 +1,75 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// EndBlockerPruneSupplierServiceConfigHistory prunes the service config history
+// of suppliers that have service config history entries that are no longer needed
+// for pending claims settlement.
```

Comment:
I believe this needs a WHY

```suggestion
// EndBlockerPruneSupplierServiceConfigHistory prunes the service config history of existing suppliers.
// If a supplier has updated its supported set of configs, but that history is no longer needed
// for various reasons (servicing relays, claim settlement, etc), it can be pruned.
// This helps reduce onchain state bloat and avoid diverting attention from non-actionable metadata.
```